sdks/java/io/tika/src/main/java/org/apache/beam/sdk/io/tika/ParseResult.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.tika;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+import org.apache.tika.metadata.Metadata;
+
+/**
+ * Tika parse result containing the file location, metadata
+ * and content converted to String.
+ */
+public class ParseResult implements Serializable {
+  private static final long serialVersionUID = 6133510503781405912L;
+  private String content;
+  private Metadata metadata;
+  private String fileLocation;
+
+  public ParseResult(String fileLocation, String content) {
+    this(fileLocation, content, new Metadata());
+  }
+
+  public ParseResult(String fileLocation, String content, Metadata metadata) {
+    this.fileLocation = fileLocation;
+    this.content = content;
+    this.metadata = metadata;
+  }
+
+  /**
+   * Gets a file content.
+   */
+  public String getContent() {
+    return content;
+  }
+
+  /**
+   * Gets a file metadata.
+   */
+  public Metadata getMetadata() {
+    return getMetadataCopy();
```

Comment:
It seems more reasonable to make a copy in the constructor, rather than create a fresh copy in the getter every time.

Reply:
Actually, if it's OK to do it in the constructor then it means a copy can simply be avoided. ParseResult owns a Metadata instance passed to it at the construction time, hence copying it does not really achieve anything, do you agree ?

Reply:
Yeah if the Metadata is not actually modified after being wrapped into the ParseResult, then it's fine to not make a copy.

Reply:
I kept making a copy in place (in case some function tries to set something there), with the assumption it will be accessed once, twice may be, and thus will won't cause any performance side-effects, while memorized the names at the construction time.

Reply:
I think it's better to be consistent with other Beam code - other code doesn't do copy-on-read for all potentially-mutable fields. It is common to do copy-on-construction, but in this case I think even that is unnecessary.